src/Files.App/Data/Items/WidgetDriveCardItem.cs

```diff
@@ -2,50 +2,61 @@
 // Licensed under the MIT License.
 
 using Microsoft.UI.Xaml.Media.Imaging;
+using Windows.Win32;
+using Windows.Win32.UI.Shell;
 
 namespace Files.App.Data.Items
 {
-	public sealed partial class WidgetDriveCardItem : WidgetCardItem, IWidgetCardItem<DriveItem>, IComparable<WidgetDriveCardItem>
+	public sealed partial class WidgetDriveCardItem : WidgetCardItem, IWidgetCardItem<IWindowsFolder>, IDisposable
 	{
-		private byte[] thumbnailData;
+		// Properties
 
-		public new DriveItem Item { get; private set; }
+		public required new IWindowsFolder Item { get; set; }
 
-		private BitmapImage thumbnail;
-		public BitmapImage Thumbnail
-		{
-			get => thumbnail;
-			set => SetProperty(ref thumbnail, value);
-		}
+		public required string Text { get; set; }
+
+		public bool ShowStorageSense => UsedSize.GigaBytes / TotalSize.GigaBytes >= Constants.Widgets.Drives.LowStorageSpacePercentageThreshold;
+
+		public bool ShowDriveUsage => TotalSize.GigaBytes > 0D;
+
+		public ByteSizeLib.ByteSize TotalSize { get; set; } = default;
+
+		public ByteSizeLib.ByteSize FreeSize { get; set; } = default;
+
+		public ByteSizeLib.ByteSize UsedSize => ByteSizeLib.ByteSize.FromBytes(TotalSize.Bytes - FreeSize.Bytes);
+
+		public string? UsageText => string.Format(Strings.DriveFreeSpaceAndCapacity.GetLocalizedResource(), FreeSize.ToSizeString(), TotalSize.ToSizeString());
```

Comment:
Are these properties based on DriveItem?

Reply:
Yes. These don't have observable setter but this is because I expect these items are cleared and reloaded when drive stuff changed or refresh is requested. TY as what Quick Access widget does too.

Reply:
Thank you for confirming.